orbit/pkg/update/disk_encryption.go

```diff
@@ -0,0 +1,41 @@
+package update
+
+import (
+	"github.com/fleetdm/fleet/v4/orbit/pkg/useraction"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+)
+
+const maxRetries = 2
+
+type DiskEncryptionRunner struct {
+	Fetcher OrbitConfigFetcher
+
+	isIdle bool
+}
+
+func ApplyDiskEncryptionRunnerMiddleware(f OrbitConfigFetcher) DiskEncryptionRunner {
+	return DiskEncryptionRunner{Fetcher: f}
+}
+
+func (d DiskEncryptionRunner) GetConfig() (*fleet.OrbitConfig, error) {
+	cfg, err := d.Fetcher.GetConfig()
+	if err != nil {
+		log.Info().Err(err).Msg("calling GetConfig from DiskEncryptionFetcher")
+		return nil, err
+	}
+
+	log.Debug().Msgf("running disk encryption fetcher middleware, notification: %v, isIdle: %v", cfg.Notifications.RotateDiskEncryptionKey, d.isIdle)
+
+	if cfg.Notifications.RotateDiskEncryptionKey && !d.isIdle {
+		d.isIdle = true
+		go func() {
+			defer func() { d.isIdle = false }()
```

Comment:
hey @mna do you have any thoughts/worries about this? is there a better way to accomplish it?

Reply:
Actually you can simplify that a bit and use `Bool.Swap`directly in the condition, e.g.:

```
if cfg.Notifications.RotateDiskEncryptionKey && !d.atomicBool.Swap(true) {
   go func() {
     defer func() { d.atomicBool.Store(false) }()
     ...
```